### PR TITLE
[Feat]: Add a demo of golang gorm

### PR DESCRIPTION
**Pull request by yyt030**

### CODE REVIEW

File: golang/gorm/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Connect OceanBase with Golang (gorm.io/driver/mysql)
+
+English | [简体中文](README-CN.md)
+
+This article describes how to connect to the OceanBase database through ``.
```
Comment (whhe) [inline golang/gorm/README.md:5]: through `` ?

File: golang/gorm/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+For details about `gorm.io/driver/mysql`, you can refer to [gorm.io/driver/mysql](https://gorm.io/driver/mysql).
+
+## Quick Start
+
+You can use `conf` to create a database connection, please refer to [gorm.io/docs/](https://gorm.io/docs) for details.
+
+Take [example.go](example.go) code as an example.
+
+
+Modify the connection info in code, and use `run.sh` to run the example code.
```
Comment (whhe) [inline golang/gorm/README.md:16]: It seems that the sample code use flags to set the connection info, you should tell the users how to set these variables by an executable example here.

**Review by whhe — COMMENTED** [review]
@yyt030 Thanks for your contribution! I left some comments. And you also need to add the gorm module in CI to ensure that the sample code is executable. 

For the `mydata` directory in applications, it looks good overall, but I think it should be added in another PR instead of being included in the current PR. Can you split it out from this PR?

**Comment by yyt030** [comment]
> @yyt030 Thanks for your contribution! I left some comments. And you also need to add the gorm module in CI to ensure that the sample code is executable.
> 
> For the `mydata` directory in applications, it looks good overall, but I think it should be added in another PR instead of being included in the current PR. Can you split it out from this PR?

I opened a new PR, please check.

**Comment by whhe** [comment]
Thanks for the update! The CI configuration for the newly added `gorm` module is still missing, could you please add it to `.github/workflows/ci.yml`? You can refer to the repository's readme file to learn how to set up a basic CI workflow.

**Comment by yyt030** [comment]
> Thanks for the update! The CI configuration for the newly added `gorm` module is still missing, could you please add it to `.github/workflows/ci.yml`? You can refer to the repository's readme file to learn how to set up a basic CI workflow.

done

**Comment by yyt030** [comment]
@whhe pls rerun ci？

**Comment by whhe** [comment]
Seems the command line arguments are not passed correctly.

```
2024/05/29 14:44:32 /home/runner/work/ob-samples/ob-samples/golang/gorm/example.go:47
[error] failed to initialize database, got error dial tcp [12](https://github.com/oceanbase/ob-samples/actions/runs/9287971069/job/25559066979?pr=56#step:7:13)7.0.0.1:3306: connect: connection refused
panic: failed to connect database

goroutine 1 [running]:
main.main()
	/home/runner/work/ob-samples/ob-samples/golang/gorm/example.go:49 +0x5a5
exit status 2
```

**Comment by yyt030** [comment]
> Seems the command line arguments are not passed correctly.
> 
> ```
> 2024/05/29 14:44:32 /home/runner/work/ob-samples/ob-samples/golang/gorm/example.go:47
> [error] failed to initialize database, got error dial tcp [12](https://github.com/oceanbase/ob-samples/actions/runs/9287971069/job/25559066979?pr=56#step:7:13)7.0.0.1:3306: connect: connection refused
> panic: failed to connect database
> 
> goroutine 1 [running]:
> main.main()
> 	/home/runner/work/ob-samples/ob-samples/golang/gorm/example.go:49 +0x5a5
> exit status 2
> ```

I install one linux with maridb server in virtualbox, and debug that then deal it  ,  connect must include ` Net`,  otherwise use 3306 default port.

**Review by whhe — APPROVED** [review]
Great work! Merging.

**Pull request merged**